pytorch_lightning/plugins/training_type/single_tpu.py

```diff
@@ -50,14 +54,27 @@ def __init__(
     def is_distributed(self) -> bool:
         return False
 
-    def setup(self) -> None:
+    def setup(self, trainer: "pl.Trainer") -> None:
+        # Revisit strategy inheritance.
         shared_params = find_shared_parameters(self.model)
         self.model_to_device()
         if is_overridden("on_post_move_to_device", self.lightning_module):
             self.model.on_post_move_to_device()
         else:
             set_shared_parameters(self.model, shared_params)
 
+        if not self.setup_optimizers_in_pre_dispatch:
+            self.setup_optimizers(trainer)
+        self.setup_precision_plugin()
+
+    def _move_optimizer_state(self, device: Optional[torch.device] = None) -> None:
```

Comment:
the device isn't used anymore. This function should be revisited across strategies.

Reply:
Issue created
https://github.com/PyTorchLightning/pytorch-lightning/issues/10820